pandas/tests/extension/test_integer.py

```diff
@@ -208,17 +208,39 @@ class TestCasting(base.BaseCastingTests):
 
 class TestGroupby(base.BaseGroupbyTests):
 
-    @pytest.mark.xfail(reason="groupby not working", strict=True)
-    def test_groupby_extension_no_sort(self, data_for_grouping):
-        super(TestGroupby, self).test_groupby_extension_no_sort(
-            data_for_grouping)
-
-    @pytest.mark.parametrize('as_index', [
-        pytest.param(True,
-                     marks=pytest.mark.xfail(reason="groupby not working",
-                                             strict=True)),
-        False
-    ])
+    @pytest.mark.parametrize('as_index', [True, False])
     def test_groupby_extension_agg(self, as_index, data_for_grouping):
-        super(TestGroupby, self).test_groupby_extension_agg(
-            as_index, data_for_grouping)
+        df = pd.DataFrame({"A": [1, 1, 2, 2, 3, 3, 1, 4],
+                           "B": data_for_grouping})
+        result = df.groupby("B", as_index=as_index).A.mean()
+        _, index = pd.factorize(data_for_grouping, sort=True)
+
+        # TODO(ExtensionIndex): remove coercion to object
+        # we don't have an easy way to represent an EA as an Index object
+        index = pd.Index(index, name="B", dtype=object)
+        expected = pd.Series([3, 1, 4], index=index, name="A")
+        if as_index:
+            self.assert_series_equal(result, expected)
+        else:
+            expected = expected.reset_index()
+            self.assert_frame_equal(result, expected)
```

Comment:
How is this one different from the base test? 
From a quick scan it looks identical, apart from the astype in `index / ...`. Is that essential?

Reply:
pls read the TODO, the issue is the coercion of the Index to object must occur currently until we support a proper EA Index.

Reply:
The diff seems to be `pd.Index(index, name="B", dtype="object")` vs. `pd.Index(index.astype('object'), name="B")`. I think the base class's test should work.

Reply:
Yes, I read the comment, I didn't get from it why those two would be different. So can you please explain a bit more in detail?

Reply:
Actually, for IntegerArray, `pd.Index` will infer that it are integers if you pass it a object array with `pd.Index(index.astype('object'), name="B")`, so that's the reason you cannot have it here? 
But I think doing `pd.Index(index, name="B", dtype="object")` in the base class will work for the other EA tests?

Reply:
Can you respond here?

Reply:
yes we need to do this:

``pd.Index(index, name="B", dtype="object")`` to avoid inferring the integers as arrays (they need to stay as object, until we have an EA Index.

However, I didn't want to change the base class as I find that slightly more generic.

Reply:
I checked out the branch, removed those tests in this file (so using the base ones), and all tests are still passing for me.

It seems that, if you keep the ExtensionArray, `pd.Index` actually does not infer the integers:

```
In [1]: a = pd.core.arrays.integer_array([1, 2, 3])

In [2]: pd.Index(a)
Out[2]: Index([1, 2, 3], dtype='object')
```

while how it was originally with converting to object ndarray, then it did. So the change you did in the base tests, were actually fine for IntegerArray as well.

Reply:
indeed, they are ok now, so removed.